modules/itemManager.ts

```diff
@@ -212,6 +210,70 @@ class ItemManager {
     await db.items.update(id, { isPurchased: !!target.checked });
     await this.populateItems();
   }
+
+  /** Toggle all items in a list as 'in cart'/purchased/selected */
+  async toggleAllListItems(event: Event) {
+    const target = event.target as HTMLInputElement;
+    const shouldMarkAll = target.checked;
+
+    const listId = this.currentListId;
+    if (!listId) return;
+    const list = await db.lists.where('id').equals(listId).toArray();
+
+    if (!list.length) return console.warn('No list with ID exists');
+    // get all the items, and mark them
+    const listItems = await db.itemLists
+      .where('listId')
+      .equals(listId)
+      .toArray();
+
+    const itemIds = listItems.map((item) => item.itemId);
+
+    // update all items
+    await db.items
+      .where('id')
+      .anyOf(itemIds)
+      .modify({ isPurchased: shouldMarkAll });
+
+    await this.populateItems();
+  }
+
+  /** Delete all items marked as purchased */
+  async removePurchasedItems() {
+    // same as usual - get the list, and the list items
+    //TODO: abstract this logic, since we do it in so many places
```

Comment:
This note should be taken care of before we merge this. 

We can create a function that returns an array of ids for items that belong to the list with the current list id. 

I try to keep PRs to 10 or less files changed (maybe that's still too many) - but while we're at it let's break the ItemManager class into a directory with different files to encapsulate big methods, like `populateItems`

Reply:
This work put me over 10 touched files (13) - but I aimed to do a few things:

- Abstract areas where we're directly hitting the database and manipulating items into a single place, the `DatabaseService` 
- Remove methods that were really just utility functions that didn't need to be on `ItemManager` at all 
- Make the ItemManager class easier to read and breakdown 

There are still areas for improvement, like the coupling of `populateItems` on all the methods. This can be fixed, but will increase changed files even more so I'll quickly do it following this.